examples/examples/runtime_update_async.rs

```diff
@@ -0,0 +1,68 @@
+/*
+	Copyright 2023 Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+//! Example that shows how to detect a runtime update and afterwards update the metadata.
+use substrate_api_client::{
```

Comment:
Would it be possible to add this example to CI? For that, an actually runtime upgrade would be needed, but I think that should be possible. See an example that does a runtime upgrade: https://github.com/scs/substrate-api-client/pull/219/files

Reply:
That would be nice and I was not aware that we already have a branch for this but I wonder if we can easily have a runtime `.wasm` that does not fail once the runtime in the parity docker gets updated. Can we set a very high version number? I'm not sure if this works.

Reply:
I believe we could use any runtime, as long as the spec_version is higher. I don't think they update the spec_version of the kitchensink runtime - and even if, if we choose something like 1000, we have quite some time :)

Current spec_version number:
https://github.com/paritytech/substrate/blob/master/bin/node/runtime/src/lib.rs#L141-L146

Reply:
I implemented it and it works quite nicely 😃 
So far I only did it for the sync example, in order to avoid duplicating all the code for triggering the runtime update but maybe it makes sense to do it. What do you think?

Reply:
Ok, I added the triggering of runtime updates now also in the async example.